pyhf/constraints.py

```diff
@@ -12,7 +10,10 @@ def __init__(self,pdfconfig):
             (pdfconfig.modifier(cname),pdfconfig.par_slice(cname))
             for cname in pdfconfig.auxdata_order
         ]
+        self.tensorlib_name = None
         self._precompute()
+        tensorlib, _ = get_backend()
+        self.tensorlib_name = tensorlib.name
```

Comment:
these two lines go into `_precompute()`. If the tensor type changes, you need to precompute, and then set the `self.tensorlib_name` again.